src/Evolution/DiscontinuousGalerkin/ProjectToBoundary.hpp

```diff
@@ -0,0 +1,246 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <type_traits>
+
+#include "DataStructures/ApplyMatrices.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/SliceIterator.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg {
+namespace detail {
+template <typename TagsList>
+struct NumberOfIndependentComponents;
+
+template <typename... Tags>
+struct NumberOfIndependentComponents<tmpl::list<Tags...>> {
+  static constexpr size_t value = (... + Tags::type::size());
+};
+}  // namespace detail
+
+/*!
+ * \brief Projects a `Variables` of volume data to a contiguous subset of
+ * a boundary `Variables`
+ *
+ * The `volume_fields` are all projected into the `face_fields` in the direction
+ * `direction`. The variables in the volume (`VolumeVarsTagsList`) must be a
```

Comment:
"The variables ... entire list must be a subset."  -> "The tags in `VolumeVarsTagsList` must be a contiguous subset of the tags in `FaceVarsTagsList`."

If you don't think that is clear enough, then add the sentence "That is, `FaceVarsTagsList` must be equivalent to `tmpl::append<Before, VolumeVarsTagsList, After>` where `Before` and `After` are `tmpl::list` of arbitrary size."